base_url/models/abstract_url.py

```diff
@@ -73,6 +74,34 @@ def _compute_is_urls_sync_required(self):
         for record in self:
             record.is_urls_sync_required = True
 
+    def _get_url_keyword(self):
+        self.ensure_one()
+        return [self.name]
+
+    def _post_process_url_key(self, key):
+        self.ensure_one()
+        return key
+
+    def _generic_compute_automatic_url_key(self):
+        if not self.ids:
```

Comment:
@sebastienbeau Your test is not right. self.ids could be a list with a mix of id and NewId. You must skip NewId when you loop on self